src/drafts/MarkdownViewer/MarkdownViewer.tsx

```diff
@@ -104,20 +97,16 @@ const MarkdownViewer = ({
     openLinksInNewTab,
   })
 
-  // If we were to inject the `...htmlContainer.children` instead of the container element itself,
-  // those children elements would be moved from the `htmlContainer` to the `outputContainer`. Then if
-  // other effects use `htmlContainer.querySelectorAll`, they wouldn't find any elements to affect
-  useEffect(() => outputContainerRef.current?.replaceChildren(htmlContainer), [htmlContainer])
-
   return loading ? (
     <Box sx={{display: 'flex', justifyContent: 'space-around', p: 2}}>
       <Spinner aria-label="Loading content..." />
     </Box>
   ) : (
     <Box
-      ref={outputContainerRef}
+      ref={htmlContainerRef}
       className="markdown-body"
       sx={{fontSize: 1, maxWidth: '100%', '& > div > :last-child': {mb: 0}}}
+      dangerouslySetInnerHTML={dangerousRenderedHTML}
     />
```

Comment:
🤔 I don't think this will work, because when the HTML changes we need to re-run the effects to reprocess the rendered DOM. But React will reuse the container in this case, so the `outputContainerRef` will not update when the `dangerouslySetInnerHTML` does